x/examples/outline-vpn-app/app/start.go

```diff
@@ -0,0 +1,87 @@
+package app
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/Jigsaw-Code/outline-sdk/x/config"
+	"github.com/Jigsaw-Code/outline-sdk/x/mobileproxy"
+	"github.com/a-h/templ"
+)
+
+var appAddress = ":8080"
+var proxyAddress = ":8181"
+var systemAddress = ":8282"
+
+var proxyDisconnectTimeoutSecond = 5
+var systemTunnelEndpoint = "${systemAddress}/tunnel/${proxyAddress}"
+
+func Start() {
+	vpn := VPNController{}
+
+	http.Handle("/", templ.Handler(serverView()))
+	http.Handle("/connection/ss://<my-shadowsocks-key>", http.HandlerFunc(vpn.handleConnectionEndpoint))
+	http.Handle("/disconnection/ss://<my-shadowsocks-key>", http.HandlerFunc(vpn.handleDisconnectionEndpoint))
+
+	http.ListenAndServe(appAddress, nil)
+}
+
+type VPNController struct {
+	proxy  *mobileproxy.Proxy
+	tunnel io.Reader
+}
+
+func (vpn *VPNController) handleConnectionEndpoint(responseWriter http.ResponseWriter, request *http.Request) {
+	proxyDialer, err := config.NewStreamDialer(request.URL.Path)
+
+	if err != nil {
+		http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	proxy, err := mobileproxy.RunProxy(proxyAddress, &mobileproxy.StreamDialer{proxyDialer})
+
+	if err != nil {
+		http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	vpn.proxy = proxy
+
+	// TODO: why this code isn't very good at all!
+	tunnel, err := os.Open("tunnel.sock")
+
+	if err != nil {
+		// do something
+	}
+
+	vpn.tunnel = tunnel
+
+	// TODO: implement system vpn tunnel service
+	// => POST /tunnel/URL forward all non-local traffic to that URL
```

Comment:
The Tunnel here is more like a system VPN configuration, typically it involves the source and the target. The source can be simple, like "all non-local TCP and UDP traffic from the system", or it can be complicated, such as "all UDP traffic destined to `8.8.8.8` from a specific app". The target would typically be represented by `IP:port` instead of a `URL`.

For the simplicity, I think we can start with `source=all TCP & UDP` and `target=<outline-server-ip>:<port>`. And we need to provide different implementations for different OS.

Reply:
Yes. At least for Windows and Linux there are no differences. But on Android and iOS, we need to adapt to the VPN API provided by the system.

Reply:
That's right.